.github/actions/templates/setEnvironmentVariables/action.yml

```diff
@@ -0,0 +1,71 @@
+#########################################################
+## 'Set environment variables' Composite Action                   ##
+#########################################################
+##
+## This composite action contains the logic to read a YAML file containing a list (dictionary) of key-value pairs and convert each key as an environment variable on the runner.
+## Example YAML File content:
+## listName:
+##  key1: 'value1'
+##  key2: 'value2'
+#########################################################
+##
+##-------------------------------------------##
+## ACTION PARAMETERS                         ##
+##-------------------------------------------##
+##
+##   |=======================================================================================================================================================================================================|
+##   | Parameter                | Required | Default | Description                                                                                      | Example                                            |
+##   |--------------------------|----------|---------|--------------------------------------------------------------------------------------------------|----------------------------------------------------|
+##   | variablesPath            | true     | ''      | The path to the YAML file that contains the key value pairs for environment variables            | './global.variables.yml'                           |
+##   |=======================================================================================================================================================================================================|
+##
+##---------------------------------------------##
+
+name: 'Set Environment Variables'
+description: 'Set Environment Variables'
+
+inputs:
+  variablesPath:
+    description: 'The path to the YAML file that contains the key value pairs for environment variables'
+    required: true
+
+runs:
+  using: 'composite'
+  steps:
+    # [Setup] task(s)
+    # ---------------
+    - name: 'Setup agent'
+      shell: pwsh
+      run: |
+        # Grouping task logs
+        Write-Output "::group::Setup agent"
+
+        # Load used functions
+        . (Join-Path $env:GITHUB_WORKSPACE 'utilities' 'pipelines' 'sharedScripts' 'Set-EnvironmentOnAgent.ps1')
+
+        # Define PS modules to install on the runner
+        $Modules = @(
+            @{ Name = 'powershell-yaml'; Version = '0.4.2'}
```

Comment:
@MrMCake @MariusStorhaug @ahmadabdalla at the beginning of the initiative we discussed about leveraging this external module a couple of times, eventually deciding not to use that, since we wanted to avoid external dependencies. This is the reason why the getInputParameters action is the way it is. Do we want to rediscuss this?

Reply:
It was, however with the variable file we now have a more complicated construct to deal with (compared to the runtime parameters of the pipeline). Maybe we need to revisit this decision. Let's have a discussion some time.